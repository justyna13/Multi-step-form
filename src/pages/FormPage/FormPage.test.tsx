import { describe } from "vitest";

describe("Form Page", () => {
  it("first test", () => {

  })
})
