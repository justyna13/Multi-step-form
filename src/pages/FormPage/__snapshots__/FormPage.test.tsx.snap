// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form Page > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="form-card"
          data-testid="multiStepForm-page-template"
        >
          <div
            class="form-wrapper"
          >
            <div
              class="form-navigation"
              data-testid="multiStepForm-navigation"
            >
              <div
                class="form-step "
                data-testid="form-step-1"
              >
                <div
                  class="form-step__number"
                >
                  1
                </div>
                <div>
                  <span
                    class="form-step__info"
                  >
                    Step 
                    1
                  </span>
                  <br />
                  <span
                    class="form-step__title"
                  >
                    Your info
                  </span>
                </div>
              </div>
              <div
                class="form-step active"
                data-testid="form-step-2"
              >
                <div
                  class="form-step__number"
                >
                  2
                </div>
                <div>
                  <span
                    class="form-step__info"
                  >
                    Step 
                    2
                  </span>
                  <br />
                  <span
                    class="form-step__title"
                  >
                    Select plan
                  </span>
                </div>
              </div>
              <div
                class="form-step "
                data-testid="form-step-3"
              >
                <div
                  class="form-step__number"
                >
                  3
                </div>
                <div>
                  <span
                    class="form-step__info"
                  >
                    Step 
                    3
                  </span>
                  <br />
                  <span
                    class="form-step__title"
                  >
                    Add-ons
                  </span>
                </div>
              </div>
              <div
                class="form-step "
                data-testid="form-step-4"
              >
                <div
                  class="form-step__number"
                >
                  4
                </div>
                <div>
                  <span
                    class="form-step__info"
                  >
                    Step 
                    4
                  </span>
                  <br />
                  <span
                    class="form-step__title"
                  >
                    Summary
                  </span>
                </div>
              </div>
            </div>
            <div
              class="form-inner"
            >
              <form
                class="multipstep-form"
                data-testid="multistep-form"
                novalidate=""
              >
                <div
                  class="form-step--second"
                  data-testid="multiStepForm-step-second"
                >
                  <div
                    class="header"
                    data-testuid="multiStepForm-step-second-ttl"
                  >
                    <h2>
                      Select your plan
                    </h2>
                    <p>
                      You have the option of monthly or yearly billing.
                    </p>
                  </div>
                  <div
                    class="form-control"
                  >
                    <div
                      class="plans-wr"
                    >
                      <div
                        class="single-plan "
                        data-testid="plan-Arcade"
                      >
                        <svg
                          height="40"
                          viewBox="0 0 40 40"
                          width="40"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                          >
                            <circle
                              cx="20"
                              cy="20"
                              fill="#FFAF7E"
                              r="20"
                            />
                            <path
                              d="M24.995 18.005h-3.998v5.998h-2v-5.998H15a1 1 0 0 0-1 1V29a1 1 0 0 0 1 1h9.995a1 1 0 0 0 1-1v-9.995a1 1 0 0 0-1-1Zm-5.997 8.996h-2v-1.999h2v2Zm2-11.175a2.999 2.999 0 1 0-2 0v2.18h2v-2.18Z"
                              fill="#FFF"
                              fill-rule="nonzero"
                            />
                          </g>
                        </svg>
                        <div>
                          <p
                            class="single-plan__ttl"
                          >
                            Arcade
                          </p>
                          <p
                            class="single-plan__desc"
                          >
                            $
                            9
                            /mo
                          </p>
                        </div>
                      </div>
                      <div
                        class="single-plan "
                        data-testid="plan-Advanced"
                      >
                        <svg
                          height="40"
                          viewBox="0 0 40 40"
                          width="40"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                          >
                            <circle
                              cx="20"
                              cy="20"
                              fill="#F9818E"
                              r="20"
                            />
                            <path
                              d="M25.057 15H14.944C12.212 15 10 17.03 10 19.885c0 2.857 2.212 4.936 4.944 4.936h10.113c2.733 0 4.943-2.08 4.943-4.936S27.79 15 25.057 15ZM17.5 20.388c0 .12-.108.237-.234.237h-1.552v1.569c0 .126-.138.217-.259.217H14.5c-.118 0-.213-.086-.213-.203v-1.583h-1.569c-.126 0-.217-.139-.217-.26v-.956c0-.117.086-.213.202-.213h1.584v-1.554c0-.125.082-.231.203-.231h.989c.12 0 .236.108.236.234v1.551h1.555c.125 0 .231.083.231.204v.988Zm5.347.393a.862.862 0 0 1-.869-.855c0-.472.39-.856.869-.856.481 0 .87.384.87.856 0 .471-.389.855-.87.855Zm1.9 1.866a.86.86 0 0 1-.87-.852.86.86 0 0 1 .87-.855c.48 0 .87.38.87.855a.86.86 0 0 1-.87.852Zm0-3.736a.862.862 0 0 1-.87-.854c0-.472.39-.856.87-.856s.87.384.87.856a.862.862 0 0 1-.87.854Zm1.899 1.87a.862.862 0 0 1-.868-.855c0-.472.389-.856.868-.856s.868.384.868.856a.862.862 0 0 1-.868.855Z"
                              fill="#FFF"
                              fill-rule="nonzero"
                            />
                          </g>
                        </svg>
                        <div>
                          <p
                            class="single-plan__ttl"
                          >
                            Advanced
                          </p>
                          <p
                            class="single-plan__desc"
                          >
                            $
                            12
                            /mo
                          </p>
                        </div>
                      </div>
                      <div
                        class="single-plan "
                        data-testid="plan-Pro"
                      >
                        <svg
                          height="40"
                          viewBox="0 0 40 40"
                          width="40"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                          >
                            <circle
                              cx="20"
                              cy="20"
                              fill="#483EFF"
                              r="20"
                            />
                            <path
                              d="M26.666 13H13.334A3.333 3.333 0 0 0 10 16.333v7.193a3.447 3.447 0 0 0 2.14 3.24c1.238.5 2.656.182 3.56-.8L18.52 23h2.96l2.82 2.966a3.2 3.2 0 0 0 3.56.8 3.447 3.447 0 0 0 2.14-3.24v-7.193A3.333 3.333 0 0 0 26.666 13Zm-9.333 6H16v1.333a.667.667 0 0 1-1.333 0V19h-1.333a.667.667 0 0 1 0-1.334h1.333v-1.333a.667.667 0 1 1 1.333 0v1.333h1.333a.667.667 0 1 1 0 1.334Zm7.333 2a2.667 2.667 0 1 1 0-5.333 2.667 2.667 0 0 1 0 5.333ZM26 18.333a1.333 1.333 0 1 1-2.667 0 1.333 1.333 0 0 1 2.667 0Z"
                              fill="#FFF"
                              fill-rule="nonzero"
                            />
                          </g>
                        </svg>
                        <div>
                          <p
                            class="single-plan__ttl"
                          >
                            Pro
                          </p>
                          <p
                            class="single-plan__desc"
                          >
                            $
                            15
                            /mo
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="plans-month-year-price-switcher"
                  >
                    <div
                      class="toggle-switch"
                    >
                      <input
                        class="toggle-switch-checkbox"
                        id="paymentPeriod"
                        type="checkbox"
                      />
                      <label
                        class="toggle-switch-label"
                        for="paymentPeriod"
                      >
                        <span
                          class="toggle-switch-label--on"
                        >
                          Yearly
                        </span>
                        <span
                          class="toggle-switch-label--off"
                        >
                          Monthly
                        </span>
                      </label>
                    </div>
                  </div>
                  <div
                    class="step-bottom"
                  >
                    <button
                      class="main-btn undefined btn-prev"
                      type="button"
                    >
                      Go Back
                    </button>
                    <button
                      class="main-btn undefined btn-next"
                      type="button"
                    >
                      Next Step
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="form-card"
        data-testid="multiStepForm-page-template"
      >
        <div
          class="form-wrapper"
        >
          <div
            class="form-navigation"
            data-testid="multiStepForm-navigation"
          >
            <div
              class="form-step "
              data-testid="form-step-1"
            >
              <div
                class="form-step__number"
              >
                1
              </div>
              <div>
                <span
                  class="form-step__info"
                >
                  Step 
                  1
                </span>
                <br />
                <span
                  class="form-step__title"
                >
                  Your info
                </span>
              </div>
            </div>
            <div
              class="form-step active"
              data-testid="form-step-2"
            >
              <div
                class="form-step__number"
              >
                2
              </div>
              <div>
                <span
                  class="form-step__info"
                >
                  Step 
                  2
                </span>
                <br />
                <span
                  class="form-step__title"
                >
                  Select plan
                </span>
              </div>
            </div>
            <div
              class="form-step "
              data-testid="form-step-3"
            >
              <div
                class="form-step__number"
              >
                3
              </div>
              <div>
                <span
                  class="form-step__info"
                >
                  Step 
                  3
                </span>
                <br />
                <span
                  class="form-step__title"
                >
                  Add-ons
                </span>
              </div>
            </div>
            <div
              class="form-step "
              data-testid="form-step-4"
            >
              <div
                class="form-step__number"
              >
                4
              </div>
              <div>
                <span
                  class="form-step__info"
                >
                  Step 
                  4
                </span>
                <br />
                <span
                  class="form-step__title"
                >
                  Summary
                </span>
              </div>
            </div>
          </div>
          <div
            class="form-inner"
          >
            <form
              class="multipstep-form"
              data-testid="multistep-form"
              novalidate=""
            >
              <div
                class="form-step--second"
                data-testid="multiStepForm-step-second"
              >
                <div
                  class="header"
                  data-testuid="multiStepForm-step-second-ttl"
                >
                  <h2>
                    Select your plan
                  </h2>
                  <p>
                    You have the option of monthly or yearly billing.
                  </p>
                </div>
                <div
                  class="form-control"
                >
                  <div
                    class="plans-wr"
                  >
                    <div
                      class="single-plan "
                      data-testid="plan-Arcade"
                    >
                      <svg
                        height="40"
                        viewBox="0 0 40 40"
                        width="40"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <circle
                            cx="20"
                            cy="20"
                            fill="#FFAF7E"
                            r="20"
                          />
                          <path
                            d="M24.995 18.005h-3.998v5.998h-2v-5.998H15a1 1 0 0 0-1 1V29a1 1 0 0 0 1 1h9.995a1 1 0 0 0 1-1v-9.995a1 1 0 0 0-1-1Zm-5.997 8.996h-2v-1.999h2v2Zm2-11.175a2.999 2.999 0 1 0-2 0v2.18h2v-2.18Z"
                            fill="#FFF"
                            fill-rule="nonzero"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="single-plan__ttl"
                        >
                          Arcade
                        </p>
                        <p
                          class="single-plan__desc"
                        >
                          $
                          9
                          /mo
                        </p>
                      </div>
                    </div>
                    <div
                      class="single-plan "
                      data-testid="plan-Advanced"
                    >
                      <svg
                        height="40"
                        viewBox="0 0 40 40"
                        width="40"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <circle
                            cx="20"
                            cy="20"
                            fill="#F9818E"
                            r="20"
                          />
                          <path
                            d="M25.057 15H14.944C12.212 15 10 17.03 10 19.885c0 2.857 2.212 4.936 4.944 4.936h10.113c2.733 0 4.943-2.08 4.943-4.936S27.79 15 25.057 15ZM17.5 20.388c0 .12-.108.237-.234.237h-1.552v1.569c0 .126-.138.217-.259.217H14.5c-.118 0-.213-.086-.213-.203v-1.583h-1.569c-.126 0-.217-.139-.217-.26v-.956c0-.117.086-.213.202-.213h1.584v-1.554c0-.125.082-.231.203-.231h.989c.12 0 .236.108.236.234v1.551h1.555c.125 0 .231.083.231.204v.988Zm5.347.393a.862.862 0 0 1-.869-.855c0-.472.39-.856.869-.856.481 0 .87.384.87.856 0 .471-.389.855-.87.855Zm1.9 1.866a.86.86 0 0 1-.87-.852.86.86 0 0 1 .87-.855c.48 0 .87.38.87.855a.86.86 0 0 1-.87.852Zm0-3.736a.862.862 0 0 1-.87-.854c0-.472.39-.856.87-.856s.87.384.87.856a.862.862 0 0 1-.87.854Zm1.899 1.87a.862.862 0 0 1-.868-.855c0-.472.389-.856.868-.856s.868.384.868.856a.862.862 0 0 1-.868.855Z"
                            fill="#FFF"
                            fill-rule="nonzero"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="single-plan__ttl"
                        >
                          Advanced
                        </p>
                        <p
                          class="single-plan__desc"
                        >
                          $
                          12
                          /mo
                        </p>
                      </div>
                    </div>
                    <div
                      class="single-plan "
                      data-testid="plan-Pro"
                    >
                      <svg
                        height="40"
                        viewBox="0 0 40 40"
                        width="40"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <circle
                            cx="20"
                            cy="20"
                            fill="#483EFF"
                            r="20"
                          />
                          <path
                            d="M26.666 13H13.334A3.333 3.333 0 0 0 10 16.333v7.193a3.447 3.447 0 0 0 2.14 3.24c1.238.5 2.656.182 3.56-.8L18.52 23h2.96l2.82 2.966a3.2 3.2 0 0 0 3.56.8 3.447 3.447 0 0 0 2.14-3.24v-7.193A3.333 3.333 0 0 0 26.666 13Zm-9.333 6H16v1.333a.667.667 0 0 1-1.333 0V19h-1.333a.667.667 0 0 1 0-1.334h1.333v-1.333a.667.667 0 1 1 1.333 0v1.333h1.333a.667.667 0 1 1 0 1.334Zm7.333 2a2.667 2.667 0 1 1 0-5.333 2.667 2.667 0 0 1 0 5.333ZM26 18.333a1.333 1.333 0 1 1-2.667 0 1.333 1.333 0 0 1 2.667 0Z"
                            fill="#FFF"
                            fill-rule="nonzero"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="single-plan__ttl"
                        >
                          Pro
                        </p>
                        <p
                          class="single-plan__desc"
                        >
                          $
                          15
                          /mo
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="plans-month-year-price-switcher"
                >
                  <div
                    class="toggle-switch"
                  >
                    <input
                      class="toggle-switch-checkbox"
                      id="paymentPeriod"
                      type="checkbox"
                    />
                    <label
                      class="toggle-switch-label"
                      for="paymentPeriod"
                    >
                      <span
                        class="toggle-switch-label--on"
                      >
                        Yearly
                      </span>
                      <span
                        class="toggle-switch-label--off"
                      >
                        Monthly
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="step-bottom"
                >
                  <button
                    class="main-btn undefined btn-prev"
                    type="button"
                  >
                    Go Back
                  </button>
                  <button
                    class="main-btn undefined btn-next"
                    type="button"
                  >
                    Next Step
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
